core/cmake/BareosCheckFunctions.cmake

```
@@ -18,82 +18,80 @@
 #   02110-1301, USA.
 
 if(NOT MSVC)
+  try_compile(
+    HAVE_CORE_SYSTEM_INTERFACES ${CMAKE_BINARY_DIR}/compile_tests
+    ${PROJECT_SOURCE_DIR}/src/compile_tests/core_system_interfaces.c
+  )
+  if(HAVE_CORE_SYSTEM_INTERFACES)
+    set(HAVE_GETPAGESIZE 1)
+    set(HAVE_POLL 1)
+    set(HAVE_READDIR_R 1)
+  else()
+    message(
+      SEND_ERROR
+        "core/src/compile_tests/core_system_interfaces.c failed to build"
+    )
```

```suggestion
  try_compile(
    HAVE_CORE_SYSTEM_INTERFACES ${CMAKE_BINARY_DIR}/compile_tests
    ${PROJECT_SOURCE_DIR}/src/compile_tests/core_system_interfaces.c
    OUTPUT_VARIABLE build_output
  )
  if(HAVE_CORE_SYSTEM_INTERFACES)
    set(HAVE_GETPAGESIZE 1)
    set(HAVE_POLL 1)
    set(HAVE_READDIR_R 1)
  else()
    message(
      SEND_ERROR
        "core/src/compile_tests/core_system_interfaces.c failed to build: ${build_output}"
    )
```

if i understand the docs right, this should help in debugging compilation issues.
```
OUTPUT_VARIABLE <var>

    Store the output from the build process in the given variable.
```